### DISCUSSION TITLE
ESP32 support

### DISCUSSION THREAD

**Discussion by AaronAutomated** [Dilbert66/esphome-dsckeybus#36] [Ideas]
Any chance for developing a yaml for ESP32 chips?

**Comment by Dilbert66** [comment]
The code already supports it. Just set the correct board and platform in the yaml and the pins in file dscalarm.h

esphome:
  name: $systemName
  platform: ESP32
  board: nodemcu-32s


Also you need to set the correct pins for the esp32 the file dscalarm.h  .  I use the pins below for the esp32


#define dscClockPin 22  // esp32 
#define dscReadPin 21   // esp32
#define dscWritePin 18  // esp32

You might also need to change the status led:

status_led:
  pin:
    number: GPIO2
    inverted: yes

**Reply by Dilbert66** [reply]
I'll post some updated files at some point in the future that makes it a little clearer to know what to change for different platforms.  I'm in the middle of adding some changes so it won't be for a while yet.
